docs/locales-process.md

```diff
@@ -0,0 +1,9 @@
+# Locales generation process
+
+Every time `yarn build` or `yarn start` is executed, the following process is
+going to be triggered as well:
+
+<img src="../static/locales.png">
+
+With this process, we reduce the amount of text we bundle on each page since we
+are querying only the necessary translations that each page needs.
```

Comment:
A bit of context on how the whole process works. Would appreciate any feedback on this.

Reply:
Also a recommended thing to read first that will help you understand and review the PR.